pkg/storegateway/bucket.go

```diff
@@ -1087,7 +1075,97 @@ func (s *BucketStore) synchronousSeriesSet(
 	s.metrics.seriesGetAllDuration.Observe(getAllDuration.Seconds())
 	s.metrics.seriesBlocksQueried.Observe(float64(len(res)))
 
-	return storepb.MergeSeriesSets(res...), cleanup, err
+	return storepb.MergeSeriesSets(res...), err
+}
+
+func (s *BucketStore) batchedSeriesSetForBlocks(
```

Comment:
Given we're calling the feature "streaming" WDYT if we reflect it in the name too?

```suggestion
func (s *BucketStore) streamingSeriesSetForBlocks(
```

Reply:
accepted your suggestion

I'm still getting confused when to use "batch," "set," or "streaming" :(